pkg/types/types.go

```diff
@@ -89,6 +89,12 @@ type Event struct {
 	Message string `json:"message,omitempty"`
 }
 
+// GetBaseEvent is needed to implement commonutils.BaseElement and
+// snapshot.SnapshotEvent interfaces.
+func (e Event) GetBaseEvent() *Event {
+	return &e
+}
+
```

Comment:
```suggestion
// GetBaseEvent is needed to implement commonutils.BaseElement and
// snapshot.SnapshotEvent interfaces.
func (*e Event) GetBaseEvent() *Event {
	return e
}
```

I think we should use a pointer receiver here as well, otherwise the struct is still being copied.

Reply:
Yeah, that would also help. However, doing so with generics became quite complex. I will let it for another PR.